core/internal/api/send.go

```diff
@@ -21,10 +22,37 @@ func (client *clientImpl) Send(req *Request) (*http.Response, error) {
 	for headerKey, headerValue := range req.Headers {
 		retryableReq.Header.Set(headerKey, headerValue)
 	}
-	client.setClientHeaders(retryableReq)
-	client.setAuthHeaders(retryableReq)
 
-	resp, err := client.retryableHTTP.Do(retryableReq)
+	return client.sendToWandbBackend(retryableReq)
+}
+
+func (client *clientImpl) Do(req *http.Request) (*http.Response, error) {
+	retryableReq, err := retryablehttp.FromRequest(req)
+	if err != nil {
+		return nil, fmt.Errorf("api: failed to parse request: %v", err)
+	}
+
+	if req.URL.Host == client.backend.baseURL.Host &&
+		strings.HasPrefix(req.URL.Path, client.backend.baseURL.Path) {
+
+		return client.sendToWandbBackend(retryableReq)
+	}
+
+	return client.sendArbitrary(retryableReq)
```

Comment:
i don't like sendArbitrary ... let's just have `send` or something

Reply:
renamed to `send`